src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -36,104 +36,64 @@
 #include <chrono>
 #include <thread>
 #include <string>
+#include <utility>
 #include <variant>
 
 #include <rust/ed25519.h>
 
 using namespace libzcash;
 
-int find_output(UniValue obj, int n) {
-    UniValue outputMapValue = find_value(obj, "outputmap");
-    if (!outputMapValue.isArray()) {
-        throw JSONRPCError(RPC_WALLET_ERROR, "Missing outputmap for JoinSplit operation");
-    }
-
-    UniValue outputMap = outputMapValue.get_array();
-    assert(outputMap.size() == ZC_NUM_JS_OUTPUTS);
-    for (size_t i = 0; i < outputMap.size(); i++) {
-        if (outputMap[i].get_int() == n) {
-            return i;
-        }
-    }
-
-    throw std::logic_error("n is not present in outputmap");
-}
-
 AsyncRPCOperation_sendmany::AsyncRPCOperation_sendmany(
-        std::optional<TransactionBuilder> builder,
-        CMutableTransaction contextualTx,
-        std::string fromAddress,
-        std::vector<SendManyRecipient> tOutputs,
-        std::vector<SendManyRecipient> zOutputs,
+        TransactionBuilder builder,
+        PaymentSource paymentSource,
+        std::vector<SendManyRecipient> recipients,
         int minDepth,
         CAmount fee,
+        bool allowRevealedAmounts,
         UniValue contextInfo) :
-        tx_(contextualTx), fromaddress_(fromAddress), t_outputs_(tOutputs), z_outputs_(zOutputs), mindepth_(minDepth), fee_(fee), contextinfo_(contextInfo)
+        builder_(builder), paymentSource_(paymentSource), recipients_(recipients),
+        mindepth_(minDepth), fee_(fee), allowRevealedAmounts_(allowRevealedAmounts),
+        contextinfo_(contextInfo)
 {
     assert(fee_ >= 0);
-
-    if (minDepth < 0) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Minconf cannot be negative");
-    }
-
-    if (fromAddress.size() == 0) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "From address parameter missing");
-    }
-
-    if (tOutputs.size() == 0 && zOutputs.size() == 0) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "No recipients");
-    }
-
-    isUsingBuilder_ = false;
-    if (builder) {
-        isUsingBuilder_ = true;
-        builder_ = builder.value();
-    }
-
-    KeyIO keyIO(Params());
-
-    useanyutxo_ = fromAddress == "ANY_TADDR";
-    if (useanyutxo_) {
-        isfromtaddr_ = true;
-    } else {
-        auto address = keyIO.DecodePaymentAddress(fromAddress);
-        if (address.has_value()) {
+    assert(mindepth_ >= 0);
```

Comment:
Why not change `minDepth` and `mindepth_` to unsigned?

Reply:
I think the reason that this was a signed value is that it's coming from UniValue::getInt, and I didn't change it.